docs/preflight/v1beta3-overview.md

```diff
@@ -0,0 +1,505 @@
+---
+title: "Preflight v1beta3: Templated Checks"
+description: "Learn how to create modular, values-driven Preflight specs with v1beta3"
+tags: ["preflight", "v1beta3", "templating"]
+sidebar_position: 4
+---
+
+## Overview
+
+Preflight v1beta3 introduces a templated, values-driven approach to authoring Preflight checks. This allows you to:
+
+- **Template your checks** using Go templates and Sprig functions
+- **Drive configuration with values files** similar to Helm charts
+- **Toggle checks on/off** based on deployment requirements
+- **Generate dynamic documentation** that reflects actual configuration
+- **Maintain reusable, modular specs** that work across environments
+
+With v1beta3, you can write a single Preflight spec that adapts to different scenarios by supplying different values files or command-line overrides.
+
+## Key Features
+
+### Templating with Values
+
+v1beta3 uses Go template syntax with the Sprig function library. Supply values via:
+- Multiple values files: `--values base.yaml --values prod.yaml`
+- Command-line overrides: `--set storage.className=fast`
+- Both combined (sets override files)
+
+```yaml
+apiVersion: troubleshoot.sh/v1beta3
+kind: Preflight
+metadata:
+  name: my-app-preflight
+spec:
+  analyzers:
+    {{- if .Values.kubernetes.version.enabled }}
+    - clusterVersion:
+        checkName: Kubernetes version
+        outcomes:
+          - fail:
+              when: '< {{ .Values.kubernetes.version.minimum }}'
+              message: Requires Kubernetes {{ .Values.kubernetes.version.minimum }}+
+          - pass:
+              message: Kubernetes version meets requirements
+    {{- end }}
+```
+
+### Self-Documenting with docStrings
+
+Every analyzer should include a `docString` that describes the requirement, rationale, and links. This can be extracted automatically for documentation:
+
+```yaml
+- docString: |
+    Title: Kubernetes Version Requirements
+    Requirement:
+      - Minimum: {{ .Values.kubernetes.version.minimum }}
+      - Recommended: {{ .Values.kubernetes.version.recommended }}
+    Ensures required APIs and security patches are available.
+    Links:
+      - https://kubernetes.io/releases/
+  clusterVersion:
+    checkName: Kubernetes version
+    outcomes:
+      - fail:
+          when: '< {{ .Values.kubernetes.version.minimum }}'
+          message: Requires at least {{ .Values.kubernetes.version.minimum }}
+      - pass:
+          message: Kubernetes version OK
+```
+
+### Dynamic Documentation
```

Comment:
How is this section different than the `Self-Documenting with docStrings` section?